### ISSUE TITLE
Add-on needs manual permissions

### ISSUE THREAD

**Issue by aryangupta701** [mozilla/geckodriver#2131]
Hello everyone, I was launching firefox using selenium (Java) with addon installed and the addon does not work till the user manually clicks on the addon button as shown in the screenshot below. For my use case I want the extension to be enabled by default is there any way to do this it will be really helpful.

![Screenshot from 2023-08-23 00-53-06](https://github.com/mozilla/geckodriver/assets/87857234/9f2c2a3c-257e-48ec-836b-521d18ed4ec9)

**Comment by thc202** [comment]
fyi, there's already https://bugzilla.mozilla.org/show_bug.cgi?id=1836309

**Comment by aryangupta701** [comment]
Thank you

**Comment by whimboo** [comment]
@aryangupta701 which extension is it that you are using here? Is it publicly available?

**Comment by aryangupta701** [comment]
ZAP browser extension. https://addons.mozilla.org/en-US/firefox/addon/zap-browser-extension/

Yes it is publicly available.

**Comment by thc202** [comment]
It's the same extension as in the referenced issue (now renamed and newer version though).

**Comment by aryangupta701** [comment]
@whimboo Do you know of any solution for this? Would be really helpful.

**Comment by whimboo** [comment]
I would need a small testcase for reproduction. If one of you could provide one as best in Python I would appreciate. Otherwise I'll try to have a look into that by next week given that this week will be kinda busy.

**Comment by aryangupta701** [comment]
I will try to create a python script for this and share with you asap. Thank you for your help.

**Comment by thc202** [comment]
@whimboo to reproduce you just need to install the extension and see that you have to manually give permissions to all sites. Did you already check https://bugzilla.mozilla.org/show_bug.cgi?id=1836309 ? afaik it's Firefox that needs (or would have to) change not `geckodriver`.

**Comment by aryangupta701** [comment]
I agree with you this is a Firefox problem. But I think while using geckodriver we have more control over firefox (like to configure how will it behave). Maybe instead of changing the firefox policies we can do some hacks using geckodriver.

**Comment by thc202** [comment]
We already went through that path and `geckodriver` does not have a feature for what we are asking, that's why we raised the issue for Firefox. If you are asking for help on how to use the profile with the custom settings, there's https://firefox-source-docs.mozilla.org/testing/geckodriver/Profiles.html

**Comment by aryangupta701** [comment]
Oh okay. Thank you.
